### ISSUE TITLE
After approving an EIP-681 "approve" request, MetaMask Mobile version 1.06 continues to display "Waiting for confirmation" after completion of transaction

### ISSUE THREAD

**Issue by ravisriv** [MetaMask/metamask-mobile#2007]
**Describe the bug**
This has reference to EIP681 feature #1847 and the items described under #1980 

MetaMask mobile (version 1.06 on iPhone X iOS 14.1) continues to show "Waiting for confirmation" message, even after an EIP681 "approve" request has been successfully executed on the blockchain (11 block confirmations are received). On certain occasions, the "approve" request does not appear immediately under the "Transaction History" (occurred twice in 4 tests). Two of the EIP681 "approve" requests, which I approved, appeared under the Transaction History after almost an hour. Both on them were showing the status 
"Approve
Submitted"
But the status of both on the Rinkeby Etherscan was "Success".

**Screenshots**
Here are the screenshots:
![image](https://user-images.githubusercontent.com/62220449/99826267-faaa5180-2b0c-11eb-94eb-6a37167d8b3e.png)
![image](https://user-images.githubusercontent.com/62220449/99826307-0a299a80-2b0d-11eb-9e3b-03184e707a9c.png)
![image](https://user-images.githubusercontent.com/62220449/99826357-1ada1080-2b0d-11eb-8231-cca2345105ec.png)

**To Reproduce**
_Steps to reproduce the behavior_
Use https://brunobar79.github.io/eip681-link-generator/# to create a deeplink for "approve" function. Scan the QR code using MetaMask mobile version 1.06.
Alternatively, you can email a request of the following type to your smart phone and click on the deeplink to invoke MetaMask. 
https://metamask.app.link/send/Token-address@Netid/approve?address=saleContractAddress&uint256=approvalAmount
Token-address is the address of the token that will be approved to be spent by the saleContract.
Netid is 4 for Rinkeby (test) and 1 for Ethereum Main (production)
saleContractAddress is the address of the saleContract, which is approved to spend tokens
approvalAmount is the amount for which "approval" is sought

**Expected behavior**
After 6 confirmations on the Rinkeby blockchain, the MetaMask mobile should stop displaying "Waiting for confirmation." The message should display "Transaction confirmed". After that under the Transaction History, it should show:
"Approve
Confirmed"
This should happen as soon as the required number of blockchain confirmations are received.

**Smartphone (please complete the following information):**
 - Device: iPhone X
 - OS: iOS 14.1
 - App Version: 1.06

-------------------------------------------------------------
_to be added after bug submission by internal support / PM_
**Severity**

- How critical is the impact of this bug on a user?
- Add stats if available on % of customers impacted
- Is this visible to all users?
- Is this tech debt?

**Comment by omnat** [comment]
Hi @ravisriv thanks for your detailed report. Could you please try to update to 1.0.7 and see if the problem exists?

**Comment by ravisriv** [comment]
Certainly, Omna! Would be a pleasure to test 1.07. Will run all the tests and report back. Thank you!

**Comment by ravisriv** [comment]
Hi @omnat - I tested MetaMask mobile v1.0.7 and it continues to have the same issue. When the EIP-681 compliant deeplink is executed from the email app on iPhone, MetaMask does not get a confirmation from the blockchain. In addition, when a QR code deeplink is scanned from MetaMask mobile v1.0.7, we have the same issue. MetaMask continues to wait for a confirmation. On the other hand, when a deeplink is executed through an app, it successful and MetaMask gives a message of "Transaction complete". Will find out what's the difference between the deeplink, which is executed the email (or QR code) and the deeplink that is executed through a mobile app. I also noticed some usability issues and 1 bug. I'll create separate items for these.

Here are the screenshots of testing with comments for anyone who is interested:
![image](https://user-images.githubusercontent.com/62220449/100565938-8d6b8e80-3279-11eb-86e2-8d6293cba2c4.png)
![image](https://user-images.githubusercontent.com/62220449/100565959-965c6000-3279-11eb-95bc-e70bd7714b4a.png)
![image](https://user-images.githubusercontent.com/62220449/100565979-a5431280-3279-11eb-81ef-5629e9457b94.png)
![image](https://user-images.githubusercontent.com/62220449/100565995-b1c76b00-3279-11eb-8644-15b6f7a3c425.png)
![image](https://user-images.githubusercontent.com/62220449/100566007-bdb32d00-3279-11eb-93d2-c23abc8227d3.png)

![image](https://user-images.githubusercontent.com/62220449/100566022-c99eef00-3279-11eb-93b5-9ba9f6d8990a.png)
![image](https://user-images.githubusercontent.com/62220449/100566047-da4f6500-3279-11eb-941f-8d74b6f46170.png)
![image](https://user-images.githubusercontent.com/62220449/100566065-e5a29080-3279-11eb-96e4-131dca07776d.png)
![image](https://user-images.githubusercontent.com/62220449/100566087-f226e900-3279-11eb-930b-b9e779b3edf4.png)
![image](https://user-images.githubusercontent.com/62220449/100566103-fce17e00-3279-11eb-81d5-937f801fd591.png)
![image](https://user-images.githubusercontent.com/62220449/100566117-05d24f80-327a-11eb-9606-3cc97052ade6.png)

**Comment by ravisriv** [comment]
Some preliminary tests show that MetaMask Mobile v.1.0.8 (build 554) on iOS 14.2 does not fix this issue. Will take a deeper look at v.1.0.8 later next week.
